webknossos/webknossos/administration/task.py

```diff
@@ -97,31 +100,56 @@ def create_from_annotations(
         assert (
             response.status_code == 200
         ), f"Failed to create tasks from files: {response.status_code}: {response.content.decode('utf-8')}"
-        result = json.loads(response.content)
-        if "warnings" in result:
-            warnings = result["warnings"]
-            if len(warnings) > 0:
-                logger.warning(
-                    f"There were {len(warnings)} warnings during task creation:"
-                )
-            for warning in warnings:
-                logger.warning(warning)
-        assert "tasks" in result, "Invalid result of task creation"
-        successes = []
-        errors = []
-        for t in result["tasks"]:
-            print(t)
-            if "success" in t:
-                successes.append(t["success"])
-            if "error" in t:
-                errors.append(t["error"])
-        if len(errors) > 0:
-            logger.error(f"{len(errors)} tasks could not be created:")
-            for error in errors:
-                logger.error(error)
-        if len(successes) > 0:
-            logger.info(f"{len(successes)} tasks were successfully created.")
-        return [cls._from_dict(t) for t in successes]
+
+        return cls._handle_task_creation_response(response)
+
+    @classmethod
+    def create(
+        cls,
+        task_type_id: str,
+        project_name: str,
+        needed_experience_domain: str,
+        needed_experience_value: int,
+        dataset_name: str,
+        starting_position: Vec3Int,
+        starting_rotation: Optional[Vec3Int] = Vec3Int(0, 0, 0),
+        instances: int = 1,
+        script_id: Optional[str] = None,
+        bounding_box: Optional[BoundingBox] = None,
+    ) -> List["Task"]:
```

Comment:
I'd move `dataset_name` up a bit, since I hope that the experience arguments might be optional in the future. I'd also add a default for the starting position, this might be fine for small datasets and is coherent with the UI.
```suggestion
    def create(
        cls,
        task_type_id: str,
        project_name: str,
        dataset_name: str,
        needed_experience_domain: str,
        needed_experience_value: int,
        starting_position: Vec3Int = Vec3Int(0, 0, 0),
        starting_rotation: Optional[Vec3Int] = Vec3Int(0, 0, 0),
        instances: int = 1,
        script_id: Optional[str] = None,
        bounding_box: Optional[BoundingBox] = None,
    ) -> List["Task"]:
```

Reply:
I thought about using a default value for the `starting_position` but discarded it soon:
1. The UI has position and rotation as required arguments
2. A good default could be the centroid of the dataset's BB. But there is no way of requesting that from the server yet.
3. Alternatively, one could use the centroid location provided by the (optional) bounding box

IMH, `0, 0, 0` is not a good default, since it will move the XY viewport in such a way that the data is only in the lower right corner - 3/4 of the screen remain black.